src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java

```diff
@@ -410,5 +401,25 @@ public NsiWrapper getNsiForOffenderByCrnAndConvictionId(
             .orElseThrow(() -> new NotFoundException(String.format("Offender with crn %s not found", crn)))
             .orElseThrow(() -> new NotFoundException(String.format("Conviction with ID %s for Offender with crn %s not found", convictionId, crn)));
     }
+
+    @GetMapping(path = "/offenders/crn/{crn}/convictions/{convictionId}/nsis/{nsiId}")
+    public Nsi getNsiByNsiId(
+            @ApiParam(name = "crn", value = "CRN for the offender", example = "A123456", required = true)
+            @NotNull @PathVariable(value = "crn") final String crn,
+            @ApiParam(name = "convictionId", value = "ID for the conviction / event", example = "2500295345", required = true)
+            @NotNull @PathVariable(value = "convictionId") final Long convictionId,
+            @ApiParam(name = "nsiId", value = "ID for the nsi", example="123456", required = true)
+            @PathVariable(value = "nsiId") Long nsiId){
+        return offenderService.getOffenderByCrn(crn)
+                .map((offender) -> convictionService.convictionsFor(offender.getOffenderId())
```

Comment:
I wonder if we should just introduce a convictionService.findConvictionForOffender(long offenderId, long convictionId) since I suspect we will repeat this again and again

Reply:
Yeah good idea, I think we already have a few instances. Will have a look